test/AElf.Runtime.WebAssembly.Tests/UnitTestExternalEnvironment.cs

```diff
@@ -0,0 +1,310 @@
+            10, 10, 10, 10, 10, 10, 10, 10,
+            10, 10, 10, 10, 10, 10, 10, 10,
+            10, 10, 10, 10, 10, 10, 10, 10,
+        });
+    }
+
+    public bool CallerIsOrigin()
+    {
+        return false;
+    }
+
+    public bool CallerIsRoot()
+    {
+        // &self.caller == &Origin::Root
+        return false;
+    }
+
+    public Address Address()
+    {
+        return WebAssemblyRuntimeTestConstants.Bob;
+    }
+
+    public long Balance()
+    {
+        return 228;
+    }
+
+    public long GetWeightPrice(Weight weight)
+    {
+        return 1312 * weight.RefTime + 103 * weight.ProofSize;
+    }
+
+    public long ValueTransferred()
+    {
+        return 1337;
+    }
+
+    byte[] IExternalEnvironment.EcdsaRecover(byte[] signature, byte[] messageHash)
+    {
+        EcdsaRecover = new Tuple<byte[], byte[]>(signature, messageHash);
+        return new ByteArrayBuilder().RepeatedBytes(3, 33);
+    }
+
+    public long Now()
+    {
+        return 1111;
+    }
+
+    public long MinimumBalance()
+    {
+        return 666;
+    }
+
+    public (byte[], long) Random(byte[] subject)
+    {
+        return (subject, 42);
+    }
+
+    public void DepositEvent(byte[] topics, byte[] data)
+    {
+        Events.Add((topics, data));
+    }
+
+    public long BlockNumber()
+    {
+        return HostSmartContractBridgeContext!.CurrentHeight;
+    }
+
+    public int MaxValueSize()
+    {
+        return 16384;
+    }
+
+    public bool AppendDebugBuffer(string message)
+    {
+        DebugMessages.Add(message);
+        return true;
+    }
+
+    public byte[] EcdsaToEthAddress(byte[] pubkey)
+    {
+        return WebAssemblyRuntimeTestConstants.Bob.ToByteArray().Take(20).ToArray();
+    }
+
+    public void SetCodeHash(Hash hash)
+    {
+        CodeHashes.Add(hash);
+    }
+
+    public int ReentranceCount()
+    {
+        return 12;
+    }
+
+    public int AccountReentranceCount(byte[] accountAddress)
+    {
+        return 12;
+    }
+
+    public void AddDelegateDependency(Hash codeHash)
+    {
+        DelegateDependencies.Add(codeHash);
+    }
+
+    public void RemoveDelegateDependency(Hash codeHash)
+    {
+        DelegateDependencies.Remove(codeHash);
+    }
+
+    public long Nonce()
+    {
+        return 995;
+    }
+
+    public void SetHostSmartContractBridgeContext(IHostSmartContractBridgeContext smartContractBridgeContext)
+    {
+        HostSmartContractBridgeContext = smartContractBridgeContext;
+    }
+
+    private string GetStateKey(byte[] key)
+    {
+        return new ScopedStatePath
+        {
+            Address = Types.Address.FromBase58("2EM5uV6bSJh6xJfZTUa1pZpYsYcCUAdPvZvFUJzMDJEx3rbioz"),
+            Path = new StatePath
+            {
+                Parts = { key.ToPlainBase58() }
+            }
+        }.ToStateKey();
+    }
+
+    private void AddExceptionToDebugMessage(Exception ex)
+    {
+        AppendDebugBuffer(ex.Message);
+#if DEBUG
```

Comment:
is this if-condition still active? or is it a comment?

Reply:
> 

Will be active if you run the test cases or publish the code via --debug mode